src/Error/Debug/ConsoleFormatter.php

```diff
@@ -0,0 +1,219 @@
+        }
+        // Windows environment checks
+        if (
+            DIRECTORY_SEPARATOR === '\\' &&
+            (!(bool)env('ANSICON') || env('ConEmuANSI') !== 'ON')
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Style text with ANSI escape codes.
+     *
+     * @param string $style The style name to use.
+     * @param string $text The text to style.
+     * @return string The styled output.
+     */
+    protected function style(string $style, string $text): string
+    {
+        $code = $this->styles[$style];
+
+        return "\033[{$code};m{$text}\033[0m";
+    }
+
+    /**
+     * Convert a tree of NodeInterface objects into a plain text string.
+     *
+     * @param \Cake\Error\Debug\NodeInterface $node The node tree to dump.
+     * @return string
+     */
+    public function dump(NodeInterface $node): string
+    {
+        $indent = 0;
+
+        return $this->_export($node, $indent);
+    }
+
+    /**
+     * Convert a tree of NodeInterface objects into a plain text string.
+     *
+     * @param \Cake\Error\Debug\NodeInterface $var The node tree to dump.
+     * @param int $indent The current indentation level.
+     * @return string
+     */
+    protected function _export(NodeInterface $var, int $indent): string
+    {
+        if ($var instanceof ScalarNode) {
+            switch ($var->getType()) {
+                case 'bool':
+                    return $this->style('const', $var->getValue() ? 'true' : 'false');
+                case 'null':
+                    return $this->style('const', 'null');
+                case 'string':
+                    return $this->style('string', "'" . (string)$var->getValue() . "'");
+                case 'int':
+                case 'float':
+                    return $this->style('visibility', "({$var->getType()})") .
+                        ' ' . $this->style('number', "{$var->getValue()}");
+                default:
+                    return "({$var->getType()}) {$var->getValue()}";
+            }
+        }
+        if ($var instanceof ArrayNode) {
+            return $this->_array($var, $indent + 1);
+        }
+        if ($var instanceof ClassNode || $var instanceof ReferenceNode) {
+            return $this->_object($var, $indent + 1);
+        }
+        if ($var instanceof SpecialNode) {
+            return $this->style('special', (string)$var->getValue());
+        }
+        throw new RuntimeException('Unknown node received ' . get_class($var));
+    }
+
+    /**
+     * Export an array type object
+     *
+     * @param \Cake\Error\Debug\ArrayNode $var The array to export.
+     * @param int $indent The current indentation level.
+     * @return string Exported array.
+     */
+    protected function _array(ArrayNode $var, int $indent): string
+    {
+        $out = $this->style('punct', '[');
+        $break = "\n" . str_repeat("  ", $indent);
+        $end = "\n" . str_repeat("  ", $indent - 1);
```

Comment:
Could we get 4 spaces of indentation?

Reply:
I went to 2 spaces to try and keep long content in narrow terminals better. I can go back to 4 though.

Reply:
That's a good point... given that everything's consistently formatted, 2 spaces aren't too bad :)